### PR TITLE
Remove ginkgo from api tests

### PR DESCRIPTION
**Pull request by tmshort**
This is related to epic #189, although there doesn't seem to be a related issue for this directory.

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Comment by tmshort** [comment]
@ncdc baby steps to removing Ginkgo...

**Review by ncdc — COMMENTED** [review]
The conditionsets package variables don't seem to be used anywhere / these tests are ... 🤷 ?

**Comment by tmshort** [comment]
> The conditionsets package variables don't seem to be used anywhere / these tests are ... 🤷 ?

@joelanford added these tests, see #122

File: api/v1alpha1/operator_types_test.go [hunk 1]
```diff
@@ -8,39 +8,51 @@ import (
 	"io/fs"
 	"strconv"
 	"strings"
-
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
+	"testing"
 
 	"github.com/operator-framework/operator-controller/internal/conditionsets"
+	// replaces with "slices" in go 1.21
```
Comment (joelanford) [inline api/v1alpha1/operator_types_test.go:14]: Nit: put this comment on the same line as the import to be more friendly with tooling that sorts/groups imports.

**Comment by ncdc** [comment]
Interesting, I haven't seen anyone validate that every condition is set post-reconcile before.

File: api/v1alpha1/operator_types_test.go [hunk 2]
```diff
@@ -8,39 +8,51 @@ import (
...
+	"golang.org/x/exp/slices"
 )
 
-var _ = Describe("OperatorTypes", func() {
-	Describe("Condition Type and Reason constants", func() {
-		It("should register types in conditionsets.ConditionTypes", func() {
-			types, err := parseConstants("Type")
-			Expect(err).NotTo(HaveOccurred())
+func TestOperatorType(t *testing.T) {
```
Comment (joelanford) [inline api/v1alpha1/operator_types_test.go:18]: Nit: better test name to convey test purpose a bit better. Same comment for the Reason tests.
```suggestion
func TestOperatorTypeRegistration(t *testing.T) {
```

**Comment by joelanford** [comment]
> Interesting, I haven't seen anyone validate that every condition is set post-reconcile before.

WDYT about that?

I set that test up as a reaction to seeing various reconcilers not properly update conditions when Reconcile doesn't follow the happy path.  This essentially forces all tested code paths to think about how all of the conditions should be set, and it somewhat guards against the possibility of multiple reconciles being necessary to fully process a change/update.

**Comment by ncdc** [comment]
In other projects, our reconcilers added conditions when they were relevant, and didn't ensure 100% of the possible conditions were present 100% of the time. I'm not against the validation, although the purist in me would rather not see exported package level variables that only exist to serve unit tests. But we don't need to debate the pros/cons of that in this PR.

**Review by ncdc — APPROVED** [review]

**Comment by tmshort** [comment]
@ncdc, I had to rebase, and handle a `lint` issue due to a comment format issue!

**Review by ncdc — APPROVED** [review]

**Pull request merged**